libs/constrained-generators/src/Constrained/Base.hs

```diff
@@ -611,6 +652,11 @@ class
   --   ∀ a. a `conformsTo` spec == a `conformsTo` constrained (\t -> toPreds t spec)
   toPreds :: Term fn a -> TypeSpec fn a -> Pred fn
 
+  -- | compute a SizeSpec, that bounds the size of what (genFromTypeSpec spec) can compute.
```

Comment:
This is confusing. Does that mean the number of elements in the set of possible values, or does it mean the size of the values themselves? The latter assumes that every type has a size, which is a major change to the model that I would expect at least shows up in some constraint somewhere. 

If it's the latter I would prefer this to be in it's own type class to avoid baking assumptions about types into `HasSpec`.

Reply:
C.f. `Forallable` and `OrdLike` for inspiration on the way to do this.